### PR TITLE
feat(batch): abort task when failed

### PR DESCRIPTION
**Pull request by BowenXiao1999**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

* Add a shutdown channel to control task abort
* The test in basic.slt.part `values(CAST('abc' AS BOOLEAN))` is a test case we can observe. It shows that it is indeed being called `abort_task`

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

File: src/frontend/src/scheduler/distributed/stage.rs [hunk 1]
```diff
@@ -53,14 +52,8 @@ const TASK_SCHEDULING_PARALLELISM: usize = 10;
...
-    Started {
-        sender: Sender<StageMessage>,
-        handle: JoinHandle<SchedulerResult<()>>,
-    },
-    Running {
-        _sender: Sender<StageMessage>,
-        _handle: JoinHandle<SchedulerResult<()>>,
-    },
+    Started,
+    Running,
```
Comment (BowenXiao1999) [inline src/frontend/src/scheduler/distributed/stage.rs:56]: Clean up the unused sender/handle here.

**Comment by BowenXiao1999** [comment]
Fixing the deterministic unittest

File: src/frontend/src/scheduler/distributed/stage.rs [hunk 2]
```diff
@@ -205,9 +203,22 @@ impl StageExecution {
         }
     }
 
-    #[expect(clippy::unused_async)]
-    pub async fn stop(&self) -> SchedulerResult<()> {
-        todo!()
+    pub async fn stop(&self) {
+        // Set state to failed.
+        {
```
Comment (liurenjie1024) [inline src/frontend/src/scheduler/distributed/stage.rs:208]: We should ignore the command here is stage already finished.

File: src/frontend/src/scheduler/distributed/stage.rs [hunk 3]
```diff
@@ -327,38 +341,62 @@ impl StageRunner {
...
+                    break;
+                }
+                status_res = all_streams.next() => {
+                        if let Some(stauts_res_inner) = status_res {
+                            // The status can be Running, Finished, Failed etc. This stream contains status from
+                            // different tasks.
+                            let status = stauts_res_inner.map_err(|e| RpcError(e.into()))?;
+                            use risingwave_pb::task_service::task_info::TaskStatus as TaskStatusProst;
+                            if TaskStatusProst::from_i32(status.task_info.as_ref().unwrap().task_status)
+                                == Some(TaskStatusProst::Running)
```
[the remark this answers is not in the document]
Comment (BowenXiao1999) [reply src/frontend/src/scheduler/distributed/stage.rs:369]: Sure. There will be more informative

File: src/frontend/src/scheduler/distributed/stage.rs [hunk 4]
```diff
@@ -327,38 +341,62 @@ impl StageRunner {
...
+        let mut shutdown_tx = shutdown_tx;
+        // This loop will stops once receive a stop message, otherwise keep processing status
+        // message.
+        loop {
+            tokio::select! {
+                    biased;
+                    _ = &mut shutdown_tx => {
+                    // Received shutdown signal from query runner, should send abort RPC to all CNs.
+                    // change state to aborted. Note that the task cancel can only happen after schedule all these tasks to CN.
+                    // This can be an optimization for future: How to stop before schedule tasks.
```
Comment (BowenXiao1999) [inline src/frontend/src/scheduler/distributed/stage.rs:358]: The abort can only happen after we schedule all tasks to CN. I think it is enough, the time before schedule task is too short so I doubt that we can stop before create task.

Comment (liurenjie1024) [reply src/frontend/src/scheduler/distributed/stage.rs:358]: LGTM

**Review by fuyufjh — APPROVED** [review]
LGTM

**Pull request merged**